### ISSUE TITLE
[Bug] Can't import colpkg backup from anki desktop

### ISSUE THREAD

**Issue by BrayanDSO** [ankidroid/Anki-Android#10388]
###### Reproduction Steps

1. Tap `Import` button
2. Select a backup from Anki Desktop
    - Demo with only 1 note [backup-2022-02-23-20.51.16.colpkg.zip](https://github.com/ankidroid/Anki-Android/files/8128616/backup-2022-02-23-20.51.16.colpkg.zip) (edit: Github doesn't accept .colpkg attachments. The .colpkg is inside the zip)

(Tested in 2.15.6 and build from master, both don't work)

###### Expected Result

It is imported sucessfully

###### Actual Result

![Screenshot_20220223-205620_AnkiDroid](https://user-images.githubusercontent.com/69634269/155430854-56354cdb-28dc-48da-b337-090873698e52.png)

###### Debug info
```
AnkiDroid Version = 2.16alpha45

Android Version = 11

Manufacturer = samsung

Model = SM-N770F

Hardware = exynos9810

Webview User Agent = Mozilla/5.0 (Linux; Android 11; SM-N770F Build/RP1A.200720.012; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/98.0.4758.101 Mobile Safari/537.36

ACRA UUID = 7b43f97d-f42d-432b-808c-dcee88875a39

Scheduler = std2

Crash Reports Enabled = false

DatabaseV2 Enabled = true
```

###### Research
*Enter an [x] character to confirm the points below:*

- [X] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [X] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [X] I have searched for similar existing issues here and on the user forum
- [X] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

**Comment by mikehardy** [comment]
why is it colpkg.zip ? Shouldn't it just be colpkg? Is it double-zipped? .colpkg files are already actually a .zip structure
:thinking:

**Comment by BrayanDSO** [comment]
My bad. That was my habit of zipping attachments. The colpkg is inside it

Em qui., 24 de fev. de 2022 09:39, Mike Hardy ***@***.***>
escreveu:

**Comment by BrayanDSO** [comment]
Github doesn't accept .colpkg attachments. It's inside it

**Comment by mikehardy** [comment]
Oh hmmm - this is important, but not nearly as important as scoped storage. And the newer database schema work and scoped storage are/were both David, I'm not sure anyone else has picked up anything in AnkiDroid-Backend besides David yet, which is fine - it's open source, people should do what they want - but I'm just stating it for context. @BrayanDSO if you wanted to start picking up that work perhaps it's a useful GSoC project although I'm not sure it was considered as such

**Comment by BrayanDSO** [comment]
I guess this is part of the Rust bridge David is building to reuse Anki desktop backend

**Comment by mikehardy** [comment]
Exactly that, although just to be extra picky (too picky?) on terms, the bridge is already built and we are already re-using the rust back-end in live versions. This is the ongoing work to support newer database schemas 

https://github.com/ankidroid/Anki-Android/issues?q=is%3Aissue+is%3Aopen+label%3ARust
https://github.com/david-allison/Anki-Android-Backend/issues

Honestly I've been meaning to get in there myself but haven't had the time to do so. David and Arthur are doing fantastic on scoped storage and that is just so important I don't want to do anything to divert their focus there (on the contrary - those are the PRs I *always* prioritize for review+merge daily with time I don't even have haha). So maybe this is a good place to put some effort in myself with any time available, so I could serve to help a bit. @TarekkMA was in there as well but I am unsure of his availability - Tarekk - no pressure at all but I'm curious if you have any thoughts + plans for AnkiDroid-Backend at the moment, or are you off doing other things?

**Comment by TarekkMA** [comment]
@mikehardy I thankfully now know some Rust and might be able to understand the backend code. I still don't know where I can start but will try to explore the codebase and see what I can do this weekend.

PS: sorry for the late reply 😅

**Comment by mikehardy** [comment]
Not sure this is still happening? If so, it *should* be fixed by #11737 because collection import will be handled by the desktop code at that point

**Comment by BrayanDSO** [comment]
Solved with the new backend
